torch/_dynamo/output_graph.py

```diff
@@ -999,6 +994,19 @@ def cleanup_graph(self):
                     self.graph.erase_node(node1)
                     self.graph.erase_node(node2)
 
+    def construct_python_bytecode(self, tx, codegen: PyCodegen, stack_values):
+        self.side_effects.codegen_hooks(codegen)
+        self.side_effects.codegen_save_tempvars(codegen)
+
+        for debug_var, args in tx.debug_locals:
+            codegen(debug_var)
+            for arg in args:
+                codegen(arg)
+            codegen.extend_output(create_call_function(len(args), True))
```

Comment:
What exactly is this doing?

Reply:
This is generating the call logging function bytecode at the end of the graph.